components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.cache.ui/src/main/resources/web/cache-mediator/update-mediator.jsp

```diff
@@ -15,83 +15,89 @@
 
-<%@ page import="org.wso2.carbon.mediator.cache.digest.DOMHASHGenerator" %>
 <%@ page import="org.wso2.carbon.mediator.cache.ui.CacheMediator" %>
 <%@ page import="org.wso2.carbon.mediator.service.ui.Mediator" %>
 <%@ page import="org.wso2.carbon.sequences.ui.util.SequenceEditorHelper" %>
 
-<%! public boolean notNullChecker(String strChecker){
+<%! public boolean notNullChecker(String strChecker) {
 
-    if(strChecker==null){
+    if (strChecker == null) {
         return false;
-    }
-    else if(strChecker!=null){
+    } else if (strChecker != null) {
         return true;
-    }
-    else if((!(strChecker.equalsIgnoreCase("")))){
+    } else if ((!(strChecker.equalsIgnoreCase("")))) {
         return true;
     }
-   return false;
+    return false;
 }%>
 <%
-    final int DEFAUTLINMEMSIZE = 1000;
     Mediator mediator = SequenceEditorHelper.getEditingMediator(request, session);
     if (!(mediator instanceof CacheMediator)) {
         // todo : proper error handling
         throw new RuntimeException("Unable to edit the mediator");
     }
     CacheMediator cacheMediator = (CacheMediator) mediator;
-    String cacheid = request.getParameter("cacheId");
-    String cacheScope = request.getParameter("cacheScope");
     String cacheType = request.getParameter("cacheType");
-    String cacheTimeout = request.getParameter("cacheTimeout");
-    String maxMsgSize = request.getParameter("maxMsgSize");
-    String impType = request.getParameter("impType");
-    String maxSize = request.getParameter("maxSize");
+    String protocolType = request.getParameter("protocolType").trim();
+    String cacheTimeout = request.getParameter("cacheTimeout").trim();
+    String maxMsgSize = request.getParameter("maxMsgSize").trim();
+    String methods = request.getParameter("methods");
+    String headersToExclude = request.getParameter("headersToExclude");
+    String responseCodes = request.getParameter("responseCodes");
+    String hashGen = request.getParameter("hashGen");
+    String maxSize = request.getParameter("maxSize").trim();
     String sequenceOption = request.getParameter("sequenceOption");
 
-    if(notNullChecker(cacheid)){
-        cacheMediator.setId(cacheid);
-    }
-    if(notNullChecker(cacheScope)){
-        cacheMediator.setScope(cacheScope.toLowerCase());
-    }
-    if(notNullChecker(cacheType)){
-        if(cacheType.equalsIgnoreCase("Collector")){
+    if (notNullChecker(cacheType)) {
+        if (cacheType.equalsIgnoreCase("Collector")) {
             cacheMediator.setCollector(true);
-        }
-        else{
+        } else {
             cacheMediator.setCollector(false);
         }
     }
-    if(notNullChecker(cacheTimeout)){
-        try{
-        cacheMediator.setTimeout(Long.parseLong(cacheTimeout));
-        }
-        catch(NumberFormatException e){
+    if (notNullChecker(cacheTimeout)) {
+        try {
+            cacheMediator.setTimeout(Long.parseLong(cacheTimeout));
+        } catch (NumberFormatException e) {
 
```

Comment:
Should we remain silent in the case of a number format exception? Shouldn't we log it and set the timeout to a default value?